packages/core/app-layout/src/components/pageHeader/AppPageHeader.vue

```diff
@@ -0,0 +1,135 @@
+<template>
+  <div class="kong-ui-app-page-header">
+    <div
+      v-if="hasBreadcrumbs"
+      class="page-header-breadcrumbs"
+      data-testid="page-header-breadcrumbs"
+    >
+      <KBreadcrumbs :items="breadcrumbs">
+        <template #divider>
+          <span class="page-header-breadcrumb-divider">/</span>
+        </template>
+      </KBreadcrumbs>
+    </div>
+
+    <div class="page-header-title-section">
+      <div class="page-header-title-wrapper">
+        <div
+          v-if="$slots['title-before']"
+          class="page-header-title-before"
+          data-testid="page-header-title-before"
+        >
+          <slot name="title-before" />
+        </div>
+        <h1
+          class="page-header-title"
+          data-testid="page-header-title"
+          :title="title"
+        >
+          {{ title }}
+        </h1>
+        <div
+          v-if="$slots['title-after']"
+          class="page-header-title-after"
+          data-testid="page-header-title-after"
+        >
+          <slot name="title-after" />
+        </div>
+      </div>
+
+      <div
+        v-if="$slots.actions"
+        class="page-header-actions"
+        data-testid="page-header-actions"
+      >
+        <slot name="actions" />
+      </div>
+    </div>
+
+    <div
+      v-if="$slots.below"
+      class="page-header-section-below"
+      data-testid="page-header-section-below"
+    >
+      <slot name="below" />
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { computed, PropType } from 'vue'
+import type { BreadcrumbItem } from '@kong/kongponents'
+
+const props = defineProps({
+  title: {
+    type: String,
+    required: true,
+  },
+  breadcrumbs: {
+    type: Array as PropType<BreadcrumbItem[]>,
+    default: () => ([]),
+  },
+})
+
+const hasBreadcrumbs = computed((): boolean => !!props.breadcrumbs?.length)
+</script>
+
+<style lang="scss" scoped>
+.kong-ui-app-page-header {
+  .page-header-breadcrumb-divider {
+    color: $kui-color-text-neutral-weak;
+  }
+
+  .page-header-title-section {
+    align-items: center;
+    display: flex;
+    gap: $kui-space-40;
+    justify-content: space-between;
+
+    .page-header-title-wrapper {
+      align-items: baseline;
+      display: flex;
+      min-width: 0;
```

Comment:
question: why min-width 0?

Reply:
This is an odd one, needed for truncation with flex. See this article @portikM found here: https://css-tricks.com/flexbox-truncated-text/

Reply:
Maybe add a comment there explaining why is that `min-width` rule there @kaiarrowood?